html_parser.js

```diff
@@ -1,144 +1,201 @@
+ * @param line  The line containing the year.
+ */
+function get_year(json, line) {
+    json.data.year = line.substring(line.search('CATALOG YEAR:') + 'CATALOG YEAR: '.length, line.search('CATALOG YEAR:') + 'CATALOG YEAR: '.length + 4);
+}
+
+/**
+ * Gets the expected graduation date of the degree audit.
+ * @param json  The json file to which this graduation date should be added.
+ * @param line  The line which contains the graduation date desired.
+ */
+function get_grad_date(json, line) {
+    json.data.grad = line.substring(line.search('GRADUATION DATE: ') + 'GRADUATION DATE: '.length, line.search('GRADUATION DATE: ') + 'GRADUATION DATE:  '.length + 7);
+}
+
+/**
+ * Gets the NUPaths associated with this degree audit.
+ * @param json  The json file to which the NUPaths should be added.
+ * @param lines The lines to be scanned for the NUPaths.
+ * @param i     The index of the line at which we currently stand.
+ */
+function get_nupaths(json, lines, i) {
+    i++;
+    while(contains(lines[i], '<br>')) {
+        var toAdd = lines[i].substring(lines[i].indexOf("(") + 1, lines[i].indexOf("(") + 3)
+        // omits bad stuff
+        if(contains(toAdd, '<') || contains(toAdd, '>')) {
+        }
+
+        else if(contains(lines[i], 'OK')|| contains(lines[i], 'IP')) {
+            json.completed.nupaths.push(toAdd);				
+        }
+        else if(contains(lines[i], 'IP')) {
+            json.inprogress.nupaths.push(toAdd);
+        }
+        else if(contains(lines[i], 'NO')) {
+            json.requirements.nupaths.push(toAdd);				
+        }
+        i++;
+    }
+
+}
+
+/**
+ * Adds the courses required by the degree audit to be taken to the JSON.
+ * @param json  The json file to which the required courses should be added.
+ * @param line  The line which contains these required courses.
+ */
+function add_courses_to_take(json, line) {
+    var courseList = line.substring(line.search('Course List') + 13).replace('<font>', '').split('<font class="auditPreviewText">').join('').split('</font>').join('TO');
+    var type = '';	
+    for(var i = 0; i < courseList.length; i++) {
+
+        var course = { };
+
+        if(!(courseList[i] == ' ' || courseList[i] == '') && contains(courseList[i], '\w\w')) {
+            course.attr = courseList[i].substring(0, 5); 
+            course.number = courseList[i].substring(5, 10);	
+            json.requirements.classes.push(course);
+        }
+
+        else {
+            course.attr = type; 
+            course.number = courseList[i];
+            json.requirements.classes.push(course);
+        }
+    }
+
+    console.log(courseList);
+}
+
+/**
+ * Adds the courses taken so far to the current JSON file.
+ * @param json  The JSON file to which the courses to be taken should be added.
+ * @param line  The line which contains the course taken.
+ */
+function add_course_taken(json, line) {
+    var course = {};
+    var courseString = line.substring(line.search('(FL|SP|S1|S2)'));
```

Comment:
var instead of let or const?

Reply:
I remember telling you they should be used instead of var awhile back... rookie mistake of me.